gantry/collection.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+import aiosqlite
+
+from gantry.models import VM, Build
+from gantry.util.gitlab import GitlabClient
+from gantry.util.prometheus import IncompleteData, PrometheusClient
+
+
+async def fetch_build(payload: dict, db: aiosqlite.Connection) -> None:
+    """
+    Fetches a job's information from Prometheus and inserts it into the database.
+    If there is data missing at any point, the function will still return so the webhook
+    responds as expected. If an exception is thrown, that behavior was unanticipated by
+    this program and should be investigated.
+
+    args:
+        payload: a dictionary containing the information from the Gitlab job hook
+        db: an active aiosqlite connection
+
+    returns: None in order to accomodate a 200 response for the webhook.
+    """
+
+    gitlab = GitlabClient()
+    prometheus = PrometheusClient()
+
+    build = Build(
+        status=payload["build_status"],
+        name=payload["build_name"],
+        id=payload["build_id"],
+        start=payload["build_started_at"],
+        end=payload["build_finished_at"],
+        retries=payload["retries_count"],
+        ref=payload["ref"],
+    )
+
+    # perform checks to see if we should collect data for this job
+    if (
+        build.status not in ("success",)
```

Comment:
```suggestion
        build.status != "success"
```

Is this a placeholder for other future statuses? If not it may be more efficient to simply compare the string.

Reply:
Yeah, that's left over from when I was originally checking for success or failure. I broke that off into #4 for now